services/QuillLMS/app/models/unit_activity.rb

```diff
@@ -157,6 +157,13 @@ def self.get_classroom_user_profile(classroom_id, user_id)
           COALESCE(cuas.locked, false) AS locked,
           COALESCE(cuas.pinned, false) AS pinned,
           MAX(acts.percentage) AS max_percentage,
+          CASE WHEN
+            unit.visible = false
+            OR ua.visible = false
+            OR cu.visible = false
+            OR acts.visible = false
```

Comment:
I don't know if this is actually simpler, but because we're looking for `false` in all of these, the four conditions here could be rewritten as:
```sql
false IN (unit.visible, ua.visible, cu.visible, acts.visible)
```